### PR TITLE
ec2: Use exponential backoff for describeinstance

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?
Be more conservative when running describeInstance on ec2 API, use exponential backup and sleep up to one minute.

### What issues does this PR fix or reference?
Fixes fails in creating instances, where the instance creation takes over 5 seconds and is therefore not done before the function runs out of retries fails.

### Previous Behavior
linear back-off with 5 retires spaced 1 second

### New Behavior
exponential back-off with 7 retries from 1 second to 64 seconds

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by gtmanfred** [comment]
This is a duplicate of #47211

File: salt/cloud/clouds/ec2.py [hunk 1]
```diff
@@ -2195,22 +2196,17 @@ def query_instance(vm_=None, call=None):
...
             )
-            attempts -= 1
-            # Just a little delay between attempts...
-            time.sleep(1)
-            continue
+        else:
+            break
 
-        break
+        time.sleep(2**attempts)
```
Comment (gtmanfred) [inline salt/cloud/clouds/ec2.py]: Can you switch this to `time.sleep(random.uniform(1, 2**attempts))` like is done in the other pr?

Thanks,
Daniel

**Review by gtmanfred — CHANGES_REQUESTED** [review]

File: salt/cloud/clouds/ec2.py (hunk 1 above)
Comment (rrroo) [reply salt/cloud/clouds/ec2.py]: Thanks for the hint. I updated the PR.

**Review by gtmanfred — APPROVED** [review]

**Comment by rrroo** [comment]
Thanks @gtmanfred and @rallytime for review and merge!
I'm proud to contribute to salt and am really happy about how quickly and friendly you help to achieve this.

**Comment by rallytime** [comment]
@rrroo Most certainly - and thank you so much for contributing! We appreciate it so much.

**Pull request merged**